File: src/main/java/com/vaadin/demo/component/numberfield/NumberFieldBasicFeatures.java
package com.vaadin.demo.component.numberfield;

import com.vaadin.flow.component.html.Span;
import com.vaadin.flow.component.icon.VaadinIcon;
import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
import com.vaadin.flow.component.textfield.NumberField;
import com.vaadin.flow.router.Route;
import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("number-field-basic-features")
public class NumberFieldBasicFeatures extends HorizontalLayout {

    public NumberFieldBasicFeatures() {
        setPadding(false);

        // tag::snippet[]
        NumberField field = new NumberField();
        field.setLabel("Label");
        field.setHelperText("Helper text");
        field.setPlaceholder("Placeholder");
        field.setTooltipText("Tooltip text");
        field.setClearButtonVisible(true);
        field.setPrefixComponent(new Span("$"));
        field.setSuffixComponent(VaadinIcon.DOLLAR.create());
        // end::snippet[]

        add(field);
    }

    public static class Exporter extends DemoExporter<NumberFieldBasicFeatures> { // hidden-source-line
    } // hidden-source-line
}

File: frontend/demo/component/numberfield/number-field-basic-features.ts
import 'Frontend/demo/init'; // hidden-source-line

import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import '@vaadin/number-field';
import '@vaadin/icon';
import '@vaadin/icons';
import '@vaadin/tooltip';
import { applyTheme } from 'Frontend/generated/theme';

@customElement('number-field-basic-features')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
        <!-- tag::snippet[] -->
        <vaadin-number-field
          label="Label"
          helper-text="Helper text"
          placeholder="Placeholder"
          clear-button-visible>
          <vaadin-tooltip slot="tooltip" text="Tooltip text"></vaadin-tooltip>
          <span slot="prefix">$</span>
          <vaadin-icon slot="suffix" icon="vaadin:dollar"></vaadin-icon>
        </vaadin-number-field>
        <!-- end::snippet[] -->
    `;
  }
}
